pwndbg/gdblib/disasm/arch.py

```diff
@@ -521,6 +523,39 @@ def _telescope_format_list(self, addresses: List[int], limit: int, emu: Emulator
                 enhance_string_len=enhance_string_len,
             )
 
+    @staticmethod
+    def _syscall_name(number: int, arch: str) -> str | None:
+        """
+        Given a syscall number and architecture, returns the name of the syscall.
+        E.g. execve == 59 on x86-64
+        """
+        arch_module = {
+            "arm": linux.arm,
+            "armcm": linux.arm,
+            "i386": linux.i386,
+            "mips": linux.mips,
+            "x86-64": linux.amd64,
+            "aarch64": linux.aarch64,
+            "rv32": linux.riscv64,
+            "rv64": linux.riscv64,
+        }.get(arch)
```

Comment:
@Ordoviz FYI I feel like there is some kind of `pwndbg.somelib.arch` or `.arch_mod` or sth that maps to it already. Otherwise maybe we should have one, or, this dict in general should probably be moved to some global scope so it is not re-created each time this is called.